perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/AppAutoRejectionScheduler.java

```diff
@@ -174,43 +200,149 @@ private void rejectExpiredApplications (List<Application> applications, Attribut
+	 *
+	 * @param attribute attribute of type map
+	 * @param lang user preferred language
+	 * @param key key used to find the value in the given attribute
+	 * @param defaultMsg returned if there is no suitable message in the attribute
+	 * @return reject message, concerning the preferred language, obtained from the given attribute,
+	 *         or the given default message
+	 */
+	private String getRejectMessage(Attribute attribute, String lang, String key, String defaultMsg) {
+		String attrKey = key + "-" + lang;
+		String attrValue = attribute != null  && attribute.getValue() != null
+				? attribute.valueAsMap().get(attrKey)
+				: null;
+
+		if (attrValue == null && attribute != null  && attribute.getValue() != null) {
+			attrValue = attribute.valueAsMap().get(key);
 		}
 
+		return attrValue != null ? attrValue : defaultMsg;
+	}
+
+	/**
+	 * Returns preferred language of user from the given application.
+	 * Preferred language is taken from the submitted application, or from the user's
+	 * preferred language attribute. If none is present, 'en' is returned.
+	 *
+	 * @param application user
+	 * @return preferred language, 'en' as default
+	 */
+	private String getUserPreferredLang(Application application) {
+		User user = application.getUser();
+		if (user == null) {
+			String appLang = null;
+			try {
+				appLang = getPreferredLangFromApplication(application);
+			} catch (PerunException e) {
+				log.error("Failed to read user preferred lang from application.", e);
+			}
+			if (appLang == null) {
+				return DEFAULT_LANG;
+			}
+			return appLang;
+		}
+		try {
+			Attribute langAttr = perun.getAttributesManagerBl().getAttribute(sess, user, A_USER_PREFERRED_LANGUAGE);
+			String attrValue = langAttr.valueAsString();
+
+			if (attrValue != null && !attrValue.isBlank()) {
+				return attrValue;
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Failed to read user preferred lang attribute.", e);
+		}
+
+		return DEFAULT_LANG;
+	}
+
+	/**
+	 * Returns preferred language filled in the application. If there is no item
+	 * with destination attribute 'urn:perun:user:attribute-def:def:preferredLanguage filled,
+	 * null is returned.
+	 *
+	 * @param application application, from which the preferred language should be returned
+	 * @return value filled by user, for application item mapping to preferredLanguage attribute,
+	 *         or null if the value is empty of there is no such item
+	 * @throws PerunException if anything goes wrong
+	 */
+	private String getPreferredLangFromApplication(Application application) throws PerunException {
+		List<ApplicationFormItemData> appData = registrarManager.getApplicationDataById(sess, application.getId());
+		for (ApplicationFormItemData data : appData) {
+			String destAttr = data.getFormItem().getPerunDestinationAttribute();
+			if (A_USER_PREFERRED_LANGUAGE.equals(destAttr) && !data.getValue().isBlank()) {
```

Comment:
Are you sure, that ``data.getValue()`` is never ``null`` (since you call ``.isBlank()`` on it)?

Reply:
I have also added a check for nullity.